pkg/segment/query/querystatus.go

```diff
@@ -171,8 +177,15 @@ func StartQuery(qid uint64, async bool) (*RunningQueryState, error) {
 func DeleteQuery(qid uint64) {
 	LogGlobalSearchErrors(qid)
 	arqMapLock.Lock()
-	delete(allRunningQueries, qid)
+	rQuery, ok := allRunningQueries[qid]
+	if ok {
+		delete(allRunningQueries, qid)
+	}
 	arqMapLock.Unlock()
+
+	if ok && !rQuery.isCancelled {
+		rQuery.timeoutCancelFunc()
```

Comment:
should we just do this cancel right inside the lock, before/after we delete from map?

Reply:
No Kunal, we do not need to do that. We can delete the qid from the map and can cancel the timeout. Calling this function is just gonna invoke `context.Done()` and then it will just exit the function completing the go routine.
Whether this `qid` is there in the map or not, it does not matter.